processing/src/main/java/io/druid/granularity/PeriodGranularity.java

```diff
@@ -45,6 +46,7 @@ public PeriodGranularity(
   )
   {
     this.period = period;
+    Preconditions.checkArgument(!Period.ZERO.equals(period), "zero period is not acceptable in QueryGranularity!");
```

Comment:
please add a test for this in QueryGranularityTest, ideally one with deserialization.